src/boosting/gbdt.cpp

```diff
@@ -421,6 +419,8 @@ bool GBDT::TrainOneIter(const score_t* gradients, const score_t* hessians) {
     models_.push_back(std::move(new_tree));
   }
 
+  ++iter_;
+
   if (!should_continue) {
```

Comment:
The one-leaf trees will be pop?

Reply:
Tested the result and stopping or using it as this is works the same (assumming there is no other built tree down the line). Incrementing the `iter_` variable is needed for the boosting to take it as a finished tree (even if the tree is of no use). Otherwise it will continue forever.

Reply:
sorry for the late response. I think we should make the behavior the same as the python-package. I remember that the one-leaf tree will be kept in the python-package, but the current implementation will pop the one-leaf trees.

Reply:
@Samsagax what do you think about my comment?